streams/src/main/java/org/apache/kafka/streams/processor/TaskId.java

```diff
@@ -59,10 +58,6 @@ public TaskId(final int topicGroupId, final int partition, final String namedTop
         }
     }
 
-    public Optional<String> namedTopology() {
```

Comment:
If we don't have this could we add something like it to `TaskMetadata`?

Reply:
Hm, `TaskMetadata` is also a public class, right? I think we'll need to do something similar in that case, ie add an internal `NamedTaskMetadata` class that exposes the `namedTopology` and then you can access it via something like `NamedTaskMetadata.extractNamedTopology(TaskMetadata)`

It's unfortunate that `TaskMetadata` has the `taskId` field as a String, rather than as an actual `TaskId` object -- not sure what the motivation there was 😕  I might try to do a very small KIP to fix that, and then we won't have to bother with this at all (you can just use the internal `NamedTaskId` class/method to access the `namedTopology`

Reply:
It would be defiantly better to just fix the taskId in taskMetadata :)

Reply:
Alright, here we are: [KIP-740: Use TaskId instead of String for the taskId field in TaskMetadata]( https://cwiki.apache.org/confluence/display/KAFKA/KIP-740%3A+Use+TaskId+instead+of+String+for+the+taskId+field+in+TaskMetadata)

That was probably the fastest/easiest KIP I've ever written